### ISSUE TITLE
Better Error for illegal variable names

### ISSUE THREAD

**Issue by gabuzi** [h5netcdf/h5netcdf#209]
#### Is your feature request related to a problem? Please describe.
I tried to save a netcdf file with h5netcdf v1.1.0, python 3.10 with a variable that ended with a forward slash `/` and encountered a cryptic error. Having a forward slash anywhere else doesn't seem to be an issue.
My very superficial research revealed that hdf5 uses forward slashes as path separators, and netcdf seems to forbid them (at least in some instances/versions), so I filed this as a feature request rather than a bug, since it may be considered illegal usage.

```
with h5netcdf.File('test', 'w') as f:
    f.dimensions = {'x': 1}
    v = f.create_variable('test/', ('x',), float)  # <= TypeError: expected bytes, str found
    v[:] = [1.]
```

#### Describe the solution you'd like
A clear error message pointing towards the illegal name. Alternatively, it would be nice if no restrictions on variable names would exist when specifying the `invalid_netcdf=True` argument, similarly to how one can use it to store complex numbers.

#### Describe alternatives you've considered
Manually encoding the slash is of course an option to avoid errors, but it doesn't solve the error-message issue.

#### Additional context
I ended up here coming from xarray and using the h5netcdf package to save my files as it seemed to be the only capable to store complex numbers (I am not currently seeking to maximize compatibility with other netCDF software, so I can live with the invalidity of the files).

xarray DataArrays are happily taking any Hashable as a name, which then leads to some incompatibility down the road when trying to solve as netCDF. For more context, however, using the scipy engine for saving netCDF from xarray does seem to tolerate trailing slashes and performs save/read without issues.

**Comment by kmuehlbauer** [comment]
Thanks @gabuzi for bringing this to attention. 

If we take the hdf5 documentation, slashes are not allowed in object names:

https://docs.hdfgroup.org/hdf5/develop/_l_b_grp_create_names.html

For h5netcdf that means that we take the string given as (see [docstring](https://h5netcdf.org/generated/h5netcdf.File.create_variable.html)):

_`Name of the new variable. If given as a path, intermediate groups will be created, if not existent.`_

In your example case the object-name would be empty as the given path is split at ('/') and the part remaining is empty.

You are completely right that we should work towards a better error message here. I'd just go with the message which would be returned by `h5py` in the same situation:

```python
import h5py
with h5py.File('test.h5', 'w') as f:
    v = f.create_dataset('test/', shape=(10,), dtype=float)
    v[:] = [1.]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[11], line 2
      1 with h5py.File('test.h5', 'w') as f:
----> 2     v = f.create_dataset('test/', shape=(10,), dtype=float)
      3     v[:] = [1.]

File ~/mambaforge/envs/h5py3/lib/python3.10/site-packages/h5py/_hl/group.py:183, in Group.create_dataset(self, name, shape, dtype, data, **kwds)
    180         parent_path, name = name.rsplit(b'/', 1)
    181         group = self.require_group(parent_path)
--> 183 dsid = dataset.make_new_dset(group, shape, dtype, data, name, **kwds)
    184 dset = dataset.Dataset(dsid)
    185 return dset

File ~/mambaforge/envs/h5py3/lib/python3.10/site-packages/h5py/_hl/dataset.py:165, in make_new_dset(parent, shape, dtype, data, name, chunks, compression, shuffle, fletcher32, maxshape, compression_opts, fillvalue, scaleoffset, track_times, external, track_order, dcpl, dapl, efile_prefix, virtual_prefix, allow_unknown_filter, rdcc_nslots, rdcc_nbytes, rdcc_w0)
    162 else:
    163     sid = h5s.create_simple(shape, maxshape)
--> 165 dset_id = h5d.create(parent.id, name, tid, sid, dcpl=dcpl, dapl=dapl)
    167 if (data is not None) and (not isinstance(data, Empty)):
    168     dset_id.write(h5s.ALL, h5s.ALL, data)

File h5py/_objects.pyx:54, in h5py._objects.with_phil.wrapper()

File h5py/_objects.pyx:55, in h5py._objects.with_phil.wrapper()

File h5py/h5d.pyx:135, in h5py.h5d.create()

ValueError: Name parameter cannot be an empty string (name parameter cannot be an empty string)
```

**Comment by kmuehlbauer** [comment]
fixed by #213
